targets/simple_switch_grpc/README.md

```diff
@@ -14,27 +14,20 @@ Github or the p4-dev mailing list.
     README](https://github.com/p4lang/PI#dependencies) to install required
     dependencies for the `--with-profo` configure flag - as well as for the
     `--with-sysrepo` configure flag if desired.
- 1. Configure, build and install PI:
+ 2. Configure, build and install PI:
     ```
     ./autogen.sh
     ./configure --with-proto --without-internal-rpc --without-cli --without-bmv2 [--with-sysrepo]
     make
     [sudo] make install
     ```
- 1. Configure and build the bmv2 code; from the root of the repository:
+ 3. Configure and build the bmv2 code from the root of the repository with `--with-pi`:
     ```
     ./autogen.sh
     ./configure --with-pi [--without-thrift] [--without-nanomsg]
     make
     [sudo] make install  # if desired
     ```
- 1. Configure and build the simple_switch_grpc code; from this directory:
-    ```
-    ./autogen.sh
-    ./configure [--with-sysrepo]
```

Comment:
Just wanted to check that subdir options such as `--with-sysrepo` work correctly.
Could you run `./configure --help=recursive` from the top-level directory and share the output here?

Reply:
```
./configure --help=recursive
Configuration of bm 1.15.0-9158ddf7:

Optional Features:
  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --disable-silent-rules  verbose build output (undo: "make V=0")
  --enable-coverage       Enable code coverage tracking
  --enable-debugger       Enable bmv2 remote debugger
  --disable-logging-macros
                          Disable compile time debug and trace logging macros
  --disable-elogger       Disable nanomsg event logger (some unit tests may
                          fail)
  --disable-undeterministic-tests
                          Skip undeterministic tests (e.g. queueing) when
                          running "make check"
  --enable-Werror         Make all compiler warnings fatal
  --enable-WP4-16-stacks  Implement stacks strictly as per the P4_16
                          specification instead of legacy behavior
  --enable-dependency-tracking
                          do not reject slow dependency extractors
  --disable-dependency-tracking
                          speeds up one-time build
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=yes]
  --enable-fast-install[=PKGS]
                          optimize for fast installation [default=yes]
  --disable-libtool-lock  avoid locking (might break parallel builds)
  --enable-modules        Allow loading third-party modules at runtime

Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
  --without-targets       Do not build targets
  --with-stress-tests     Include stress tests
  --with-pdfixed          Build pdfixed for bmv2
  --with-nanomsg          Support generating Nanomsg events
  --with-thrift           Build Thrift RPC service, if disabled then you must
                          have some other way of controlling the switch
  --with-pi               Build PI implementation for bmv2, this
                          implementation is not the one included in the PI
                          repository (which is the recommended one)
  --with-pic[=PKGS]       try to use only PIC/non-PIC objects [default=use
                          both]
  --with-aix-soname=aix|svr4|both
                          shared library versioning (aka "SONAME") variant to
                          provide on AIX, [default=aix].
  --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
  --with-sysroot[=DIR]    Search for dependent libraries within DIR (or the
                          compiler's sysroot if not specified).

Some influential environment variables:
  PYTHON      the Python interpreter
  CXX         C++ compiler command
  CXXFLAGS    C++ compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CC          C compiler command
  CFLAGS      C compiler flags
  LT_SYS_LIBRARY_PATH
              User-defined run-time library search path.
  CPP         C preprocessor
  CXXCPP      C++ preprocessor

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to <antonin@barefootnetworks.com>.

Configuration of simple-switch-grpc 0.1:

Optional Features:
  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --disable-silent-rules  verbose build output (undo: "make V=0")
  --enable-dependency-tracking
                          do not reject slow dependency extractors
  --disable-dependency-tracking
                          speeds up one-time build
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=yes]
  --enable-fast-install[=PKGS]
                          optimize for fast installation [default=yes]
  --disable-libtool-lock  avoid locking (might break parallel builds)
  --enable-Werror         Make all compiler warnings fatal

Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
  --with-pic[=PKGS]       try to use only PIC/non-PIC objects [default=use
                          both]
  --with-aix-soname=aix|svr4|both
                          shared library versioning (aka "SONAME") variant to
                          provide on AIX, [default=aix].
  --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
  --with-sysroot[=DIR]    Search for dependent libraries within DIR (or the
                          compiler's sysroot if not specified).
  --with-sysrepo          Use sysrepo gNMI service implementation [default=no]
  --with-thrift           Enable runtime Thrift server [default=no]

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CXX         C++ compiler command
  CXXFLAGS    C++ compiler flags
  PYTHON      the Python interpreter
  LT_SYS_LIBRARY_PATH
              User-defined run-time library search path.
  CPP         C preprocessor
  CXXCPP      C++ preprocessor
  PKG_CONFIG  path to pkg-config utility
  PKG_CONFIG_PATH
              directories to add to pkg-config's search path
  PKG_CONFIG_LIBDIR
              path overriding pkg-config's built-in search path
  PROTOBUF_CFLAGS
              C compiler flags for PROTOBUF, overriding pkg-config
  PROTOBUF_LIBS
              linker flags for PROTOBUF, overriding pkg-config
  GRPC_CFLAGS C compiler flags for GRPC, overriding pkg-config
  GRPC_LIBS   linker flags for GRPC, overriding pkg-config

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to <antonin@barefootnetworks.com>.
```

Reply:
```
./configure --with-pi --with-sysrepo
...

Features recap ......................
With Sysrepo .................. : yes
With Thrift ................... : no

Features recap ......................
Coverage enabled .............. : no
Logging macros enabled ........ : yes
With Nanomsg .................. : yes
Event logger enabled .......... : yes
Debugger enabled .............. : no
With Thrift ................... : yes
With pdfixed .................. : no
With PI ....................... : yes
```